vault/eventbus/bus.go

```diff
@@ -65,6 +95,22 @@ func (bus *EventBus) Send(ctx context.Context, eventType logical.EventType, data
 	return err
 }
 
+func (bus *EventBus) WithPlugin(namespace *namespace.Namespace, eventPluginInfo *logical.EventPluginInfo) *pluginEventBus {
+	return &pluginEventBus{
+		bus:        bus,
+		namespace:  namespace,
+		pluginInfo: eventPluginInfo,
+	}
+}
+
+// Send sends an event to the event bus and routes it to all relevant subscribers.
+// This function does *not* wait for all subscribers to acknowledge before returning.
+func (bus *pluginEventBus) Send(ctx context.Context, eventType logical.EventType, data *logical.EventData) error {
+	ctx = namespace.ContextWithNamespace(ctx, bus.namespace)
```

Comment:
We should definitely get some automated tests in place to exercise this in the enterprise repo.

Reply:
Once this lands I'll definitely do some more namespace related tests in enterprise